File: mobile/lib/models/register_dto.dart
class RegisterDto {
  final String email;
  final String password;
  final String matricula; // Matricula o RFC
  final String nombre;
  final String role;

  RegisterDto({
    required this.email,
    required this.password,
    required this.matricula,
    required this.nombre,
    required this.role,
  });

  Map<String, dynamic> toJson() {
    return {
      'Email': email,
      'Password': password,
      'Matricula': matricula,
      'Nombre': nombre,
      'role': role,
    };
  }
} 

File: ApiCrud/ApiCrud/DTOs/RegisterDto.cs
using System.ComponentModel.DataAnnotations;

namespace ApiCrud.DTOs
{
    public class RegisterDto
    {
        
        public string Email { get; set; }
        public string Password { get; set; }

        [Required]
        public string role { get; set; }

        [MaxLength(14)]
        public string? Matricula { get; set; }
        [MaxLength(100)]
        public string? Nombre { get; set; }
    }
}
